### ISSUE TITLE
PIN cache should use the kernel keyring

### ISSUE THREAD

**Issue by Foxboron** [Foxboron/ssh-tpm-agent#74]

**Comment by dcousens** [comment]
is this the same as using the [Kernel Key Retention Service](https://www.kernel.org/doc/html/v6.0/security/keys/core.html#kernel-key-retention-service)?

**Comment by Foxboron** [comment]
Yes! This seems more like the internal docs. I've only read over the syscalls as implemented by the `x/sys` package.
